services/skus/service.go

```diff
@@ -1329,129 +1328,151 @@ func (s *Service) verifyCredential(ctx context.Context, req credential, w http.R
-			ExpiresAt string `json:"expiresAt"`
-			Token     string `json:"token"`
+	if err != nil {
+		// for time limited v2 credentials we expose a credential id so the caller can decide whether to allow multiple redemptions
+		if err.Error() == cbr.ErrDupRedeem.Error() && req.GetType(ctx) == timeLimitedV2 {
+			return handlers.RenderContent(ctx, &RedemptionResult{ID: decodedCredential.TokenPreimage, Duplicate: true}, w, http.StatusOK)
 		}
-
-		var bytes []byte
-		bytes, err = base64.StdEncoding.DecodeString(req.GetPresentation(ctx))
-		if err != nil {
-			logger.Error().Err(err).
-				Msg("failed to decode the request token presentation")
-			return handlers.WrapError(err, "Error in decoding presentation", http.StatusBadRequest)
+		// if this is a duplicate redemption these are not verified
+		if err.Error() == cbr.ErrDupRedeem.Error() || err.Error() == cbr.ErrBadRequest.Error() {
+			return handlers.WrapError(err, "invalid credentials", http.StatusForbidden)
 		}
-		logger.Debug().Str("presentation", string(bytes)).Msg("presentation decoded")
+		return handlers.WrapError(err, "Error verifying credentials", http.StatusInternalServerError)
+	}
 
-		var presentation Presentation
-		err = json.Unmarshal(bytes, &presentation)
-		if err != nil {
-			logger.Error().Err(err).
-				Msg("failed to unmarshal the request token presentation")
-			return handlers.WrapError(err, "Error in presentation formatting", http.StatusBadRequest)
-		}
+	return handlers.RenderContent(ctx, &RedemptionResult{ID: decodedCredential.TokenPreimage, Duplicate: false}, w, http.StatusOK)
+}
 
-		logger.Debug().Str("presentation", string(bytes)).Msg("presentation unmarshalled")
+// verifyTimeLimitedV1Credential - given a time limited v1 credential, verify it.
+func (s *Service) verifyTimeLimitedV1Credential(ctx context.Context, req credential, w http.ResponseWriter) *handlers.AppError {
+	logger := logging.Logger(ctx, "verifyTimeLimitedV1Credential")
+	// Presentation includes a token and token metadata test test
+	type Presentation struct {
+		IssuedAt  string `json:"issuedAt"`
+		ExpiresAt string `json:"expiresAt"`
+		Token     string `json:"token"`
+	}
 
-		// Ensure that the credential being redeemed (opaque to merchant) matches the outer credential details
-		issuerID, err := encodeIssuerID(req.GetMerchantID(ctx), req.GetSku(ctx))
-		if err != nil {
-			logger.Error().Err(err).
-				Msg("failed to encode the issuer id")
-			return handlers.WrapError(err, "Error in outer merchantId or sku", http.StatusBadRequest)
-		}
-		logger.Debug().Str("issuer", issuerID).Msg("issuer encoded")
+	var bytes []byte
+	var err error
+	bytes, err = base64.StdEncoding.DecodeString(req.GetPresentation(ctx))
+	if err != nil {
+		logger.Error().Err(err).
+			Msg("failed to decode the request token presentation")
+		return handlers.WrapError(err, "Error in decoding presentation", http.StatusBadRequest)
+	}
+	logger.Debug().Str("presentation", string(bytes)).Msg("presentation decoded")
```

Comment:
same as [can we remove debug logging](https://github.com/brave-intl/bat-go/pull/1995/files#r1331720525)